### ISSUE TITLE
Drop support for node 16

### ISSUE THREAD

**Issue by genaris** [hyperledger/aries-framework-javascript#1510]
Should we:
- make the wrappers only support node 18 (they basically already do)
- update afj to only support 18 (and maybe 20) in 0.5
- add the patched ref napi to all the wrappers by default

Would solve quite some performance and setup issues.

Node 16 will be EOL in september, we can be ahead of EOL by a few months for once as we have a good reason

_Originally posted by @TimoGlastra in https://github.com/hyperledger/aries-framework-javascript/issues/1508#issuecomment-1639811295_

**Comment by genaris** [comment]
I agree that this would help us on simplifying the setup and the issues that users have found. It will be also a good moment to switch the test suites to use askar-based wallets instead of indy-sdk.

Have you tested the patched ref-napi in node 20? We have extensively used it under node 18.x (including last releases) but not the chance to use it under 20 yet.

**Comment by blu3beri** [comment]
I think dropping support for 16 is fine. Users can still override it if we set it with the `engine` field inside the `package.json` but it shows clear intent we do not support 16 officially.

**Comment by TimoGlastra** [comment]
Discussed on AFJ WG. There are some issues in Node 18 when using Windows Subshell for Linux 2, which are not an issue in Node 16.

Will address that before we drop support for Node 16

@dnaicker86 will open an issue to describe issues encountered.

No further concerns for dropping support for Node 16

**Comment by dnaicker86** [comment]
> Discussed on AFJ WG. There are some issues in Node 18 when using Windows Subshell for Linux 2, which are not an issue in Node 16.
> 
> Will address that before we drop support for Node 16
> 
> @dnaicker86 will open an issue to describe issues encountered.
> 
> No further concerns for dropping support for Node 16

Logged issue https://github.com/hyperledger/aries-framework-javascript/issues/1522 and published code for testing install with different versions of node 16 or 18 for windows/linux https://github.com/dnaicker86/aries-js-project-1

**Comment by genaris** [comment]
Just an update: based on what we found in https://github.com/hyperledger/aries-framework-javascript/issues/1522#issuecomment-1646925186, it was not a problem of node version but command line processor. So it seems that everything is fine about dropping node 16.
